### PR TITLE
platform: xilinx: update timer

### PR DESCRIPTION
**Pull request by amiclaus**
- use platform ops
- add timer get elapsed func

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
v2:
- fix code style

File: drivers/platform/xilinx/xilinx_timer.c [hunk 1]
```diff
@@ -423,3 +424,31 @@ int32_t no_os_timer_count_clk_set(struct no_os_timer_desc *desc,
 
 	return 0;
 }
+
+/**
+ * @brief Get the elapsed time in nsec for the timer.
+ * @param [in] desc         - Pointer to the device handler.
+ * @param [in] elapsed_time - The elapsed time in nsec.
+ * @return 0 in case of success, error code otherwise.
```
Comment (CiprianRegus) [inline drivers/platform/xilinx/xilinx_timer.c:432]: This only returns `-ENOSYS`.

File: projects/ad7768-evb/src.mk [hunk 2]
```diff
@@ -47,6 +47,7 @@ SRC_DIRS += $(DRIVERS)/axi_core/axi_adc_core
 SRCS += $(DRIVERS)/api/no_os_gpio.c \
         $(DRIVERS)/api/no_os_i2c.c  \
         $(DRIVERS)/api/no_os_irq.c  \
+	$(DRIVERS)/api/no_os_timer.c \
```
Comment (CiprianRegus) [inline projects/ad7768-evb/src.mk:50]: Is this required, since this project doesn't use timers? I guess `no_os_timer.h` is also included unnecessarily.

**Review by CiprianRegus — APPROVED** [review]
Looks good, minor stuff.

**Review by amiclaus — COMMENTED** [review]
v3:
- change return comment to `return -ENOSYS`

**Review by buha — APPROVED** [review]

**Pull request merged**